### ISSUE TITLE
Add option to show tab-close-button on each tab

### ISSUE THREAD

**Issue by yafp** [Croydon/vertical-tabs-reloaded#106]
I would love to see an option in the preference to enable or disable a permanent displayed tab-close button on all tabs (not just on the currently active one)

Add-on version: 0.8.3.1
Firefox version: 54.0
OS + OS-Version: Ubuntu 16.04
Other tab + user interface related add-ons:

**Comment by ludasmokva** [comment]
This is curious to me. 
If you mouse over any tab, the tab-close option appears. So you don't actually have to 'click' on a tab to have this option available.
Just wondering why you would like it to be permanently visible? Maybe I missed something.

**Comment by yafp** [comment]
For me the current situation feels like it delays the process: i have to target the right side of the tab, then wait until the close-image appears and can then activate the event.

Maybe it is not measurable .... but sometimes i do have the impression that the actual process of loading the close-image itself is delayed as well (while it isnt always.... might be load related)

**Comment by ludasmokva** [comment]
Ahh... ok.
For me the process is much much quicker. The 'close' button appears instantaneously; I go to close and the button is there. 
It could be your browser is just slow. FF really lagged on my last system - not sure why. Everything seemed to be going through a 'delay' of sorts before executing. Not sure how vertical tabs responded as I didn't have it installed, but I imagine it would have been something like you described. 
I think maybe some of my older add-ons were incompatible - or just slowing the system down rather than the comp being dated (although that was an issue as well).

**Comment by ZimM-LostPolygon** [comment]
I have 700 tabs opened, and indeed, there is a delay before the "close" button is shown. The result is, I cannot close a whole bunch of tabs by just clicking on the "close" button without moving the cursor, because half of the time the tab gets selected instead, meaning that the close button wasn't there at the time of the click. I've managed to hack the theme CSS to get the desired behavior, but that's hardly a reliable solution.

**Comment by justdave** [comment]
I'm having the opposite problem...  I'm accidentally closing tabs because the close button isn't there, and I happened to click near the edge of the tab trying to switch to it.  And it closes instead because I nailed the close button I couldn't see.

**Comment by Croydon** [comment]
Reference: https://github.com/Croydon/vertical-tabs-reloaded/issues/97#issuecomment-322272573

**Comment by grahamperrin** [comment]
Enabling this option will have the added benefit of reducing (or ceasing) moving targets – shifting texts – when (with e.g. [Stylish](https://addons.mozilla.org/addon/stylish/)) there's a user preference for **a close button on the left of each tab** in desktop environments that prefer things to the right: 

```
@namespace url("http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul");

/* https://support.mozilla.org/questions/1157451#answer-961582 */
/* https://support.mozilla.org/questions/1157451#answer-961663 */

.tabbrowser-tab .tab-close-button {
display:-moz-box!important;
-moz-box-ordinal-group: 0 !important;
margin-left: -2px !important;
margin-right: 3px !important;
}
 
.tabbrowser-tab .tab-throbber,
.tabbrowser-tab .tab-icon-image,
.tabbrowser-tab .tab-sharing-icon-overlay,
.tabbrowser-tab .tab-icon-overlay,
.tabbrowser-tab .tab-label-container,
.tabbrowser-tab .tab-icon-sound {
-moz-box-ordinal-group: 2 !important;
}
```

**Comment by Croydon** [comment]
Added as experimental feature in 0.10.0a5.

**It has huge obvious bugs when used with so many tabs that a scrollbar is displayed. Don't enable the feature if you don't understand what _experimental_ means.**


![alwaysdisplayclose](https://user-images.githubusercontent.com/1593194/37628123-4a5ceb84-2bd8-11e8-83a3-e4b304fdc911.PNG)

**Comment by Croydon** [comment]
The bugs remaining are coming from more generic layout issues I couldn't solve so far.

Please don't expect this feature as stable anytime soon.

If someone with better CSS skills wants to help out, contact me please! 😄

**Comment by Croydon** [comment]
Fixed by https://github.com/Croydon/vertical-tabs-reloaded/commit/b668679a5b6f17542708e799d8e01566f17835b9

**Comment by Croydon** [comment]
Will be available as a stable feature in 0.11.0.
